File: CS/GaugePropertyEditor.Web/StateIndicatorGaugePropertyEditor.cs
using System;
using DevExpress.XtraGauges.Core.Model;
using System.Drawing;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Editors;
using DevExpress.Web.ASPxGauges;
using DevExpress.Web.ASPxGauges.Gauges.State;
using DevExpress.XtraGauges.Base;
using DevExpress.Web.ASPxGauges.Gauges;

namespace GaugePropertyEditor.Web {
    [PropertyEditor(typeof(Enum))]
    public class StateIndicatorGaugePropertyEditor : WebGaugePropertyEditorBase {
        public StateIndicatorGaugePropertyEditor(Type objectType, DictionaryNode info)
            : base(objectType, info) {
        }
        public override void SetupGaugeCore(ASPxGaugeControl gaugeContainer) {
            StateIndicatorGauge siGauge = (StateIndicatorGauge)gaugeContainer.AddGauge(GaugeType.StateIndicator);
            StateIndicatorComponent stateIndicator = siGauge.AddIndicator();
            stateIndicator.BeginUpdate();
            BindableComponent = stateIndicator;
            StateIndicatorShapeType[] shapes = new StateIndicatorShapeType[] {
                        StateIndicatorShapeType.TrafficLight2,
                        StateIndicatorShapeType.TrafficLight3,
                        StateIndicatorShapeType.TrafficLight4,
                    };
            stateIndicator.States.Clear();
            foreach (StateIndicatorShapeType shape in shapes) {
                IndicatorStateWeb state = new IndicatorStateWeb(shape);
                stateIndicator.States.Add(state);
            }
            stateIndicator.Size = new SizeF(100, 200);
        }
        protected override object GetControlValueCore() {
            return ((StateIndicatorComponent)BindableComponent).StateIndex;
        }
        protected override void ReadValueCore() {
            ((StateIndicatorComponent)BindableComponent).StateIndex = (int)PropertyValue;
        }
    }
}

File: VB/GaugePropertyEditor.Web/StateIndicatorGaugePropertyEditor.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports DevExpress.Web.ASPxGauges
Imports DevExpress.XtraGauges.Base
Imports DevExpress.ExpressApp.Editors
Imports DevExpress.XtraGauges.Core.Model
Imports DevExpress.Web.ASPxGauges.Gauges
Imports DevExpress.Web.ASPxGauges.Gauges.State

    <PropertyEditor(GetType(System.Enum), False)> _
     Public Class StateIndicatorGaugePropertyEditor
        Inherits WebGaugePropertyEditorBase
        Public Sub New(ByVal objectType As Type, ByVal info As DevExpress.ExpressApp.Model.IModelMemberViewItem)
            MyBase.New(objectType, info)
        End Sub
        Public Overrides Sub SetupGaugeCore(ByVal gaugeContainer As ASPxGaugeControl)
            Dim siGauge As StateIndicatorGauge = CType(gaugeContainer.AddGauge(GaugeType.StateIndicator), StateIndicatorGauge)
            Dim stateIndicator As StateIndicatorComponent = siGauge.AddIndicator()
            stateIndicator.BeginUpdate()
            BindableComponent = stateIndicator
            Dim shapes() As StateIndicatorShapeType = {StateIndicatorShapeType.TrafficLight2, StateIndicatorShapeType.TrafficLight3, StateIndicatorShapeType.TrafficLight4}
            stateIndicator.States.Clear()
            For Each shape As StateIndicatorShapeType In shapes
                Dim state As New IndicatorStateWeb(shape)
                stateIndicator.States.Add(state)
            Next shape
            stateIndicator.Size = New SizeF(100, 200)
        End Sub
        Protected Overrides Function GetControlValueCore() As Object
            Return (CType(BindableComponent, StateIndicatorComponent)).StateIndex
        End Function
        Protected Overrides Sub ReadValueCore()
            CType(BindableComponent, StateIndicatorComponent).StateIndex = Convert.ToInt32(PropertyValue)
        End Sub
    End Class
